python/foglamp/services/core/api/north.py

```diff
@@ -48,16 +70,25 @@
 
     schedules = []
     schedule_list = await server.Server.scheduler.get_schedules()
+    latest_tasks = await _get_tasks_status()
     for sch in schedule_list:
         if sch.name in north_schedules:
+            task = latest_tasks.get(sch.name, None)
             schedules.append({
                 'id': str(sch.schedule_id),
                 'name': sch.name,
                 'processName': sch.process_name,
                 'repeat': sch.repeat.total_seconds() if sch.repeat else 0,
                 'day': sch.day,
                 'enabled': sch.enabled,
-                'exclusive': sch.exclusive
+                'exclusive': sch.exclusive,
+                'taskStatus': None if task is None else {
+                    'state': [t.name.capitalize() for t in list(Task.State)][int(task['state']) - 1],
+                    'startTime': str(task['start_time']),
+                    'endTime': str(task['end_time']),
+                    'exitCode': task['exit_code'],
+                    'reason': task['reason'],
```

Comment:
exit_code and reason keys for debug?

Reply:
There is no change in `tasks` table  for reason and exit code yet so far, but once those are set by north egress, REST API will return them as is.